### PR TITLE
Update the NWB reader to extract units from the files

### PR DESCRIPTION
**Pull request by DrTaDa**

### CODE REVIEW

**Comment by DrTaDa** [comment]
@anilbey There isn't. I updated the code to get the units form the file rather than asking it from the user.

**Comment by anilbey** [comment]
Ah great!

File: bluepyefe/reader.py [hunk 1]
```diff
@@ -199,75 +248,43 @@ def nwb_reader_BBP(in_data):
     _check_metadata(
         in_data,
         nwb_reader_BBP.__name__,
-        ["filepath", "i_unit", "v_unit", "t_unit", "protocol_name"],
+        ["filepath", "protocol_name"],
     )
 
     filepath = in_data["filepath"]
-    r = h5py.File(filepath, "r")
+    content = h5py.File(filepath, "r")
```
Comment (anilbey) [inline bluepyefe/reader.py:255]: It'd be good to close the h5 file afterwards. Not calling `content.close()` method can cause corruption to the h5 file.
Another alternative is to open it inside a context manager. This also makes sure the file gets closed even if there's a runtime exception.
`with h5py.File(filepath, "r") as content:`

Comment (wvangeit) [reply bluepyefe/reader.py:255]: Yeah, I'd go for the context manager.

Comment (DrTaDa) [reply bluepyefe/reader.py]: Done !

**Pull request merged**